File: Views/Main/HomeContentView.swift
import SwiftUI

struct HomeContentView: View {
    @EnvironmentObject var appState: AppState
    @Binding var activeSheet: SheetType?
    
    // Vérifier si un partenaire est connecté
    private var hasConnectedPartner: Bool {
        guard let partnerId = appState.currentUser?.partnerId else { return false }
        return !partnerId.isEmpty
    }
    
    var body: some View {
        NavigationView {
            ZStack {
                // Fond gris clair pour toute la vue
                Color(red: 0.97, green: 0.97, blue: 0.98)
                    .ignoresSafeArea(.all)
                
                // ScrollView avec section distance partenaires et grille des catégories
                ScrollView {
                    VStack(spacing: 30) {
                        // Section distance entre partenaires (remplace le logo)
                        PartnerDistanceView(
                            onPartnerAvatarTap: {
                                activeSheet = .partnerManagement
                            },
                            onDistanceTap: { showPartnerMessageOnly in
                                if showPartnerMessageOnly {
                                    activeSheet = .partnerLocationMessage
                                } else {
                                    activeSheet = .locationPermission
                                }
                            }
                        )
                            .environmentObject(appState)
                            .padding(.top, 100) // Ajouter plus d'espace depuis la status bar
                        
                        // Section invitation partenaire (si pas connecté)
                        if !hasConnectedPartner {
                            PartnerInviteView {
                                activeSheet = .partnerManagement
                            }
                            .padding(.top, -15) // Rapprocher de la section distance
                        }
                        
                        // Liste des catégories (style rectangulaire)
                        VStack(spacing: 20) {
                            // Utiliser toutes les catégories - le FreemiumManager gère l'accès
                            ForEach(Array(QuestionCategory.categories.enumerated()), id: \.element.id) { index, category in
                                CategoryListCardView(category: category) {
                                    print("🔥🔥🔥 HOMECONTENTVIEW CALLBACK: Catégorie sélectionnée: \(category.title)")
                                    activeSheet = .questions(category)
                                    print("🔥🔥🔥 HOMECONTENTVIEW CALLBACK: activeSheet = .questions(\(category.title))")
                                }
                                .environmentObject(appState)
                                
                                // Ajouter le sous-titre premium après la première catégorie (Toi et moi / en-couple)
                                if index == 0 {
                                    if let subtitle = getPremiumCategoriesSubtitle() {
                                        Text(subtitle)
                                            .font(.caption)
                                            .foregroundColor(.secondary)
                                            .multilineTextAlignment(.center)
                                            .padding(.horizontal, 20)
                                            .padding(.top, 10)
                                            .padding(.bottom, 10)
                                    }
                                }
                            }
                        }
                        .padding(.horizontal, 20)
                        
                        // Section Widgets
                        VStack(alignment: .leading, spacing: 16) {
                            // Titre "Widgets" à gauche
                            HStack {
                                Text(NSLocalizedString("widgets", comment: "Widgets title"))
                                    .font(.system(size: 22, weight: .bold))
                                    .foregroundColor(.black)
                                    .padding(.horizontal, 20)
                                
                                Spacer()
                            }
                            
                            // Carte widget (remplace la section widgets défilants)
                            WidgetPreviewSection(onWidgetTap: {
                                print("📱 HomeContentView: Carte widget tappée, ouverture de la page widgets")
                                activeSheet = .widgets
                            })
                        }
                        
                        // Section Statistiques sur le couple
                        CoupleStatisticsView()
                            .environmentObject(appState)
                            .padding(.top, 30)
                    }
                    .padding(.bottom, 100) // Espace pour le menu du bas
                    .background(
                        // Dégradé rose en arrière-plan du contenu
                        VStack {
                            LinearGradient(
                                gradient: Gradient(colors: [
                                    Color(hex: "#FD267A").opacity(0.3),
                                    Color(hex: "#FD267A").opacity(0.1),
                                    Color.white.opacity(0)
                                ]),
                                startPoint: .top,
                                endPoint: .bottom
                            )
                            .frame(height: 350) // Augmenter la hauteur pour couvrir plus d'espace
                            .ignoresSafeArea(edges: .top)
                            
                            Spacer()
                        }
                        , alignment: .top
                    )
                }
                .ignoresSafeArea(edges: .top) // Permettre au ScrollView de remonter jusqu'en haut
            }
        }
        .navigationBarHidden(true)
    }
    
    // MARK: - Premium Categories Subtitle Helper
    
    /// Helper pour obtenir le sous-titre des catégories premium
    private func getPremiumCategoriesSubtitle() -> String? {
        // Seulement si l'utilisateur a un partenaire connecté
        guard hasConnectedPartner else {
            return nil
        }
        
        // Vérifier le statut d'abonnement
        if appState.currentUser?.isSubscribed ?? false {
            return NSLocalizedString("premium_categories_subscribed", comment: "Premium categories subtitle for subscribed users")
        } else {
            return NSLocalizedString("premium_categories_not_subscribed", comment: "Premium categories subtitle for non-subscribed users")
        }
    }
} 

File: Views/Main/HomeContentView.kt
@file:Suppress("unused")

package com.yourapp.ui.home

import android.content.Context
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.MutableState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.yourapp.R

/**
 * Kotlin/Compose conversion of SwiftUI `HomeContentView`.
 *
 * Notes:
 * - iOS `EnvironmentObject` -> explicit parameter [appState].
 * - iOS `@Binding var activeSheet: SheetType?` -> [activeSheet] as a hoisted [MutableState].
 * - `NSLocalizedString(...)` -> Compose `stringResource(R.string.key)` (or Context.getString in helpers).
 * - Layout: ZStack -> Box, ScrollView -> Column + verticalScroll.
 * - Gradient header reproduced with a top overlay Box.
 */
@Composable
fun HomeContentView(
    appState: AppState,
    activeSheet: MutableState<SheetType?>,
    modifier: Modifier = Modifier
) {
    val context = LocalContext.current

    val hasConnectedPartner by remember(appState.currentUser?.partnerId) {
        val partnerId = appState.currentUser?.partnerId.orEmpty()
        androidx.compose.runtime.mutableStateOf(partnerId.isNotEmpty())
    }

    Box(
        modifier
            .fillMaxSize()
            // Swift: Color(red: 0.97, green: 0.97, blue: 0.98)
            .background(Color(0xFFF7F7FA))
    ) {
        // --- Pink top gradient background (height 350dp) ---
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(350.dp)
                .align(Alignment.TopCenter)
                .background(
                    brush = Brush.verticalGradient(
                        colors = listOf(
                            Color(0xFFFD267A).copy(alpha = 0.30f),
                            Color(0xFFFD267A).copy(alpha = 0.10f),
                            Color.Transparent
                        )
                    )
                )
        )

        // --- Scrollable content ---
        Column(
            modifier = Modifier
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(bottom = 100.dp) // Space for bottom navigation
        ) {
            // Partner distance section (replaces logo)
            PartnerDistanceView(
                onPartnerAvatarTap = {
                    activeSheet.value = SheetType.PartnerManagement
                },
                onDistanceTap = { showPartnerMessageOnly ->
                    activeSheet.value = if (showPartnerMessageOnly) {
                        SheetType.PartnerLocationMessage
                    } else {
                        SheetType.LocationPermission
                    }
                }
            )
                .then(Modifier.padding(top = 100.dp)) // More space from status bar

            // Partner invite section if not connected
            if (!hasConnectedPartner) {
                PartnerInviteView(onInviteTap = {
                    activeSheet.value = SheetType.PartnerManagement
                })
                    .then(Modifier.padding(top =  (-15).dp)) // pull closer to distance card
            }

            // Category list (rectangular style). Freemium access handled elsewhere.
            Column(Modifier.padding(horizontal = 20.dp)) {
                QuestionCategory.categories.forEachIndexed { index, category ->
                    CategoryListCardView(
                        category = category,
                        onClick = {
                            println("🔥 HomeContentView: Catégorie sélectionnée: ${'$'}{category.title}")
                            activeSheet.value = SheetType.Questions(category)
                        }
                    )
                    if (index < QuestionCategory.categories.lastIndex) {
                        Spacer(modifier = Modifier.height(20.dp))
                    }
                }
            }

            // Widgets section
            Column(Modifier.fillMaxWidth()) {
                Row(Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
                    Text(
                        text = stringResource(id = R.string.widgets),
                        style = MaterialTheme.typography.titleLarge.copy(fontWeight = FontWeight.Bold),
                        modifier = Modifier.padding(horizontal = 20.dp)
                    )
                    Spacer(Modifier.weight(1f))
                }

                WidgetPreviewSection(onWidgetTap = {
                    println("📱 HomeContentView: Carte widget tappée, ouverture de la page widgets")
                    activeSheet.value = SheetType.Widgets
                })
            }

            // Couple statistics
            CoupleStatisticsView(appState = appState)
                .then(Modifier.padding(top = 30.dp))
        }
    }
}

// --- Helper: Premium categories subtitle (kept for parity with Swift) ---
fun getPremiumCategoriesSubtitle(context: Context, appState: AppState): String? {
    val isSubscribed = appState.currentUser?.isSubscribed ?: false
    return if (isSubscribed) {
        context.getString(R.string.premium_categories_subscribed)
    } else {
        context.getString(R.string.premium_categories_not_subscribed)
    }
}

// -----------------------------
// Models & placeholders
// -----------------------------

/** Minimal user + app state to match usage in this screen */
data class User(
    val id: String? = null,
    val partnerId: String? = null,
    val isSubscribed: Boolean = false,
)

class AppState(
    val currentUser: User? = null,
    // Add other services as needed (journalService, favoritesService, etc.)
)

/** Categories used on the Home screen */
@Suppress("MemberVisibilityCanBePrivate")
data class QuestionCategory(
    val id: String,
    val title: String, // You can switch to @StringRes if you store titles in strings.xml
) {
    companion object {
        // Provide your real categories here
        val categories: List<QuestionCategory> = emptyList()
    }
}

/** Hoisted destination/sheet types (parity with Swift `SheetType`) */
sealed class SheetType {
    data class Questions(val category: QuestionCategory) : SheetType()
    object PartnerManagement : SheetType()
    object PartnerLocationMessage : SheetType()
    object LocationPermission : SheetType()
    object Widgets : SheetType()
}

// -----------------------------
// UI child placeholders (replace with your real implementations)
// -----------------------------

@Composable
fun PartnerDistanceView(
    onPartnerAvatarTap: () -> Unit,
    onDistanceTap: (showPartnerMessageOnly: Boolean) -> Unit,
    modifier: Modifier = Modifier,
) {
    // TODO: Replace with your real composable
    Spacer(modifier.height(1.dp))
}

@Composable
fun PartnerInviteView(
    onInviteTap: () -> Unit,
    modifier: Modifier = Modifier,
) {
    // TODO: Replace with your real composable
    Spacer(modifier.height(1.dp))
}

@Composable
fun CategoryListCardView(
    category: QuestionCategory,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
) {
    // TODO: Replace with your real composable
    Spacer(modifier.height(120.dp))
}

@Composable
fun WidgetPreviewSection(
    onWidgetTap: () -> Unit,
    modifier: Modifier = Modifier,
) {
    // TODO: Replace with your real composable
    Spacer(modifier.height(140.dp))
}

@Composable
fun CoupleStatisticsView(
    appState: AppState,
    modifier: Modifier = Modifier,
) {
    // TODO: Replace with your real composable
    Spacer(modifier.height(100.dp))
}
